File: test_patch.py
import urllib.request
import urllib.error

req = urllib.request.Request(
    'http://127.0.0.1:8000/sessions/ses_0000009c231e890b8Aniw2bOnR',
    data=b'{"mode":"edit"}',
    headers={'Content-Type': 'application/json'},
    method='PATCH'
)

try:
    res = urllib.request.urlopen(req)
    val = res.read().decode()
    with open('test_out.txt', 'w') as f:
        f.write("OK: " + val)
except urllib.error.HTTPError as e:
    with open('test_out.txt', 'w') as f:
        f.write("ERR: " + str(e.code) + " " + e.read().decode())
except Exception as e:
    with open('test_out.txt', 'w') as f:
        f.write("EXC: " + str(e))

File: test_patch.js
const fetch = require('node-fetch') || fetch;
async function go() {
    const res = await fetch('http://127.0.0.1:8000/sessions/ses_0000009c231e890b8Aniw2bOnR', {
        method: 'PATCH',
        body: JSON.stringify({ mode: 'edit' }),
        headers: { 'Content-Type': 'application/json' }
    });
    const text = await res.text();
    console.log(`STATUS: ${res.status}`);
    console.log(`BODY: ${text}`);
}
go().catch(console.error);
